source/backend/api/Helpers/Healthchecks/PimsGeoserverHealthcheck.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Pims.Core.Exceptions;
+
+namespace Pims.Api.Helpers.Healthchecks
+{
+    public class PimsGeoserverHealthCheck : IHealthCheck
+    {
+        private readonly IConfiguration _proxyOptions;
+        private readonly string _url;
+        private readonly int _statusCode;
+
+        public PimsGeoserverHealthCheck(IConfiguration section)
+        {
+            _proxyOptions = section.GetSection("Geoserver");
+            _url = section.GetValue<string>("HealthChecks:Geoserver:Url");
+            _statusCode = section.GetValue<int>("HealthChecks:Geoserver:StatusCode");
+        }
+
+        public PimsGeoserverHealthCheck(IConfiguration section, string url, int statusCode)
+        {
+            _proxyOptions = section;
+            _url = url;
+            _statusCode = statusCode;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            using (var client = new HttpClient())
+            {
+                try
+                {
+                    var basicAuth = Convert.ToBase64String(Encoding.GetEncoding("ISO-8859-1").GetBytes($"{_proxyOptions.GetValue<string>("ServiceUser")}:{_proxyOptions.GetValue<string>("ServicePassword")}"));
```

Comment:
Will log some tech debt for this - currently all of the proxy logic is directly in the controller for geoserver - as a result the auth logic needs to be reproduced here. Would be nice to have that in a dependency injectable service so that the healthcheck can reuse the exact logic used by the endpoint (this puts this implementation at risk of drifting from the functional implementation).